src/agni/server/manager/UserManager.java

```diff
@@ -16,8 +16,69 @@ public UserManager(InfoSender infoSender,
     }
 
     @Override
-    public void userRequest(String ip, byte type) {
+    public void userRequest(String ip, byte type, String message) {
         // TODO Auto-generated method stub
+    	String username = userDataGuard.getUsername(ip); 
+        if(type == 0x00) { 			// join chat; message = group to join
+        	String groupName = message; 
+        	if (userDataGuard.groupExists(groupName)) {
+        		userDataGuard.addUserToChat(username,groupName); 
+        		infoSender.sendInfo(ip, "success: joining " + groupName);
+        	}
+        	else {
+        		infoSender.sendInfo(ip, "failed: " + groupName + " doesn't exist");
+        	}
+        }
+        else if (type == 0x01){ 	// leave chat 
+        	String groupName = message; 
```

Comment:
User shouldn't need to specify what chat they're leaving. Users will only be part of one chat, so they will leave the one they're currently in.